test/sai_test/sai_rif_test.py

```diff
@@ -37,43 +37,48 @@ def setUp(self):
 
     def test_ingress_mac_update(self):
         """
-        Generate Packets, with SIP:192.168.0.1 DIP:192.168.11.1 DMAC:SWITCH_MAC
-        Send packet on Port1
-        Verify packet received on one of the LAG1's member
-        Set RIF mac to MacX, the RIF related to Port1
-        Send packet on Port1
-        Verify no packet was received on any LAG1 member
+        1. Generate Packets, with ``SIP:192.168.0.1`` ``DIP:10.1.1.100`` ``DMAC:SWITCH_MAC``
+        2. Send packet on Port5
+        3. Verify packet received on one of the LAG1's member
+        4. Set RIF mac to ``MacX``, the RIF related to Port5 VLAN interface
+        5. Send packet on Port5 
+        6. Verify no packet was received on any LAG1 member
         """
         print("\nmacUpdateTest()")
 
         new_router_mac = "00:77:66:55:44:44"
-        ip_dst = self.servers[11][1].ipv4
+        src_dev = self.servers[0][1]
+        target_dev = self.t1_list[1][100]
+        send_port = self.dut.port_obj_list[5]
+        recv_dev_ports = self.get_dev_port_indexes(
+            list(filter(lambda item: item != 1, target_dev.l3_lag_obj.member_port_indexs)))
+
         pkt = simple_tcp_packet(eth_dst=ROUTER_MAC,
-                                eth_src=self.servers[0][1].mac,
-                                ip_dst=ip_dst,
-                                ip_src=self.servers[0][1].ipv4,
+                                eth_src=src_dev.mac,
+                                ip_dst=target_dev.ipv4,
+                                ip_src=src_dev.ipv4,
                                 ip_id=105,
                                 ip_ttl=64)
 
-        exp_pkt = simple_tcp_packet(eth_dst=self.servers[11][1].l3_lag_obj.neighbor_mac,
+        exp_pkt = simple_tcp_packet(eth_dst=target_dev.l3_lag_obj.neighbor_mac,
                                     eth_src=ROUTER_MAC,
-                                    ip_dst=ip_dst,
-                                    ip_src=self.servers[0][1].ipv4,
+                                    ip_dst=target_dev.ipv4,
+                                    ip_src=src_dev.ipv4,
                                     ip_id=105,
                                     ip_ttl=63)
 
-        send_packet(self, 1, pkt)
-        verify_packet_any_port(self, exp_pkt, self.servers[11][1].l3_lag_obj.member_port_indexs)
+        send_packet(self, send_port.dev_port_index, pkt)
+        verify_packet_any_port(self, exp_pkt, recv_dev_ports)
 
         print("Updating src_mac_address to %s" % (new_router_mac))
         sai_thrift_set_router_interface_attribute(
-            self.client, self.dut.port_obj_list[0].rif_list[-1], src_mac_address=new_router_mac)
+            self.client, self.dut.vlans[10].rif_list[-1], src_mac_address=new_router_mac)
```

Comment:
0 is the vlan set in common config, we should use 0 here, please help correct it